### ISSUE TITLE
Build failing with latest Angular 11 and @types/mapbox-gl 

### ISSUE THREAD

**Issue by almothafar** [Wykks/ngx-mapbox-gl#256]
Build failing with latest Angular 11 and `@types/mapbox-gl`, with error: 

```
TS2422: A class can only implement an object type or intersection of object types with statically known members.
```
I got typescript 4, Angular 11, `@types/mapbox-gl` 1.12.8

![image](https://user-images.githubusercontent.com/13887564/99520966-789b1b00-299c-11eb-994e-a806a373d76d.png)

![image](https://user-images.githubusercontent.com/13887564/99520864-52757b00-299c-11eb-89e7-af4d9a65c43f.png)

with `@types/mapbox-gl` version 1.12.7 got no problem.

**Comment by dmytro-gokun** [comment]
@almothafar Just use @types/mapbox-gl 1.12.7 for the time being. At the moment I do not have time to fix this. Another possibility is to submit a PR.

**Comment by Kcorab** [comment]
This issue has nothing to do with angular 11. You could simply remove the `Layer` interface in your `layer.component.d.ts`.
Please insert always the text of your error message, so a web search engine is able to find the issue.

**Comment by dmytro-gokun** [comment]
So, the root cause of this issue is this PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/49611
It introduced a breaking change to the type definitions. I've now asked the PR author to explain why he had to do that. Let's wait a bit for the answer and then decide what to do here. Ideally, the type definition has to be fixed. If that does not happen, I will think how to fix the library compilation with smallest possible impact on its users.

**Comment by dmytro-gokun** [comment]
@almothafar This should be fixed now when using @types/mapbox-gl >= 1.12.9
